test/contracts/let_patterns.aes

```diff
@@ -0,0 +1,13 @@
+contract LetPatterns =
```

Comment:
There doesn't seem to be a test case for the match support in list comprehensions.

Reply:
Good catch. I added it to my temp test file instead of the actual test case.